DCOS/autoscale-testing/AutoScale.Tests.ps1

```diff
@@ -0,0 +1,271 @@
+﻿$here = Split-Path -Parent $MyInvocation.MyCommand.Path
+
+Import-Module AzureRM
+
+function Login-AzureRmFromEnv {
+    if ($Env:CLIENT_ID -eq $null) {
+        if (Test-Path "$here\env.ps1") {
+            . "$here\env.ps1"
+        }
+        else {
+            Write-Error "Could not find $here\env.ps1 file to source credentials, please create it"
+            exit 1
+        }
+    }
+    
+    # Suppress error message for check
+    $ErrorActionPreference = 'silentlycontinue'
+    
+    # Check if already logged in
+    try {
+        $subscription = Get-AzureRmSubscription
+    }
+    catch {
+        $subscription = $null
+    }
+    
+    $ErrorActionPreference = 'continue'
+    
```

Comment:
You can replace this entire block:
```
# Suppress error message for check
$ErrorActionPreference = 'silentlycontinue'

# Check if already logged in
try {
    $subscription = Get-AzureRmSubscription
}
catch {
    $subscription = $null
}

$ErrorActionPreference = 'continue'
```
by simply:
```
$subscription = Get-AzureRmSubscription -ErrorAction SilentlyContinue
```
The error action `SilentlyContinue` will apply only to the `Get-AzureRmSubscription` command.

If the command fails, the script won't break and the `$subscription` variable will be `$null`, just as intended.